File: server/features/worktree.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

// WorktreeList describes the list-worktrees RPC.
var WorktreeList = featureregistry.Feature{
	ID:          "worktree-list",
	Title:       "List Worktrees",
	Description: "Lists all git worktrees available for a given repository path.",
	RPCIDs:      []string{"worktree:list"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(WorktreeList)
}

File: web-app/src/lib/features/features/worktree.ts
import type { Feature } from '../types';

export const worktreeFeatures = {
  'worktree-list': {
    id: 'worktree-list',
    title: 'List Worktrees',
    description: 'Lists all git worktrees available for a given repository path.',
    rpcIds: ['worktree:list'],
    componentPaths: [],
    testIds: [
      'TestListWorktrees_EmptyPath',
      'TestListWorktrees_NonGitPath',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;
